argo/providers/kfp/provider.go

```diff
@@ -398,24 +402,32 @@ func ConnectToKfpApi(address string) (*GrpcKfpApi, error) {
 	}, nil
 }
 
-func (kfpp KfpProvider) EventingServer(ctx context.Context, providerConfig KfpProviderConfig) (generic.EventingServer, error) {
+func (kfpp KfpProvider) EventingServer(ctx context.Context, provider string, namespace string) (generic.EventingServer, error) {
 	k8sClient, err := CreateK8sClient()
 	if err != nil {
 		return nil, err
 	}
 
-	metadataStore, err := ConnectToMetadataStore(providerConfig.GrpcMetadataStoreAddress)
+	config := &KfpProviderConfig{
+		Name: provider,
+	}
+
+	if err = LoadProvider[KfpProviderConfig](ctx, k8sClient, provider, namespace, config); err != nil {
```

Comment:
could the provider config be returned from `LoadProvider` explicitly? Bit confusing atm

Reply:
It's the go way, mutation. The config object needs to be created here as one is VAI and one is KFP specific, could just return as well as mutating in the function but then we would need to add the extra lines of if statement to handle the error response.